lib/request.js

```diff
@@ -120,13 +120,15 @@ internals.Request.prototype._read = function (size) {
 
     setImmediate(() => {
 
+        /* $lab:coverage:off$ */
         if (this._shot.isDone) {
             if (this._shot.simulate.end !== false) {        // 'end' defaults to true
```

Comment:
Can you wrap only this if statement please? We should have coverage on the outer if